src/analysis/p_series.lean

```diff
@@ -0,0 +1,169 @@
+      (nat.le_of_lt_succ $ (Ico.mem.mp hk).2),
+  convert sum_le_sum this,
+  simp [pow_succ, two_mul]
+end
+
+lemma sum_condensed_le (hf : ∀ ⦃m n⦄, 1 < m → m ≤ n → f n ≤ f m) (n : ℕ) :
+  (∑ k in range (n + 1), (2 ^ k) •ℕ f (2 ^ k)) ≤ f 1 + 2 •ℕ ∑ k in Ico 2 (2 ^ n + 1), f k :=
+begin
+  convert add_le_add_left (nsmul_le_nsmul_of_le_right (sum_condensed_le' hf n) 2) (f 1),
+  simp [sum_range_succ', add_comm, pow_succ, mul_nsmul, sum_nsmul]
+end
+
+end finset
+
+namespace ennreal
+
+variable {f : ℕ → ennreal}
+
+lemma le_tsum_condensed (hf : ∀ ⦃m n⦄, 0 < m → m ≤ n → f n ≤ f m) :
+  (∑' k, f k) ≤ f 0 + ∑' k : ℕ, (2 ^ k) * f (2 ^ k) :=
+begin
+  rw [ennreal.tsum_eq_supr_nat' (nat.tendsto_pow_at_top_at_top_of_one_lt _root_.one_lt_two)],
+  refine supr_le (λ n, (finset.le_sum_condensed hf n).trans (add_le_add_left _ _)),
+  simp only [nsmul_eq_mul, nat.cast_pow, nat.cast_two],
+  apply ennreal.sum_le_tsum
+end
+
+lemma tsum_condensed_le (hf : ∀ ⦃m n⦄, 1 < m → m ≤ n → f n ≤ f m) :
+  (∑' k : ℕ, (2 ^ k) * f (2 ^ k)) ≤ f 1 + 2 * ∑' k, f k :=
+begin
+  rw [ennreal.tsum_eq_supr_nat' (tendsto_at_top_mono nat.le_succ tendsto_id), two_mul, ← two_nsmul],
+  refine supr_le (λ n, le_trans _ (add_le_add_left (nsmul_le_nsmul_of_le_right
+    (ennreal.sum_le_tsum $ finset.Ico 2 (2^n + 1)) _) _)),
+  simpa using finset.sum_condensed_le hf n
+end
+
+end ennreal
+
+namespace nnreal
+
+/-- Cauchy condensation test for a series of `nnreal` version. -/
+lemma summable_condensed_iff {f : ℕ → ℝ≥0} (hf : ∀ ⦃m n⦄, 0 < m → m ≤ n → f n ≤ f m) :
+  summable (λ k : ℕ, (2 ^ k) * f (2 ^ k)) ↔ summable f :=
+begin
+  simp only [← ennreal.tsum_coe_ne_top_iff_summable, ne.def, not_iff_not, ennreal.coe_mul,
+    ennreal.coe_pow, ennreal.coe_two],
+  split; intro h,
+  { replace hf : ∀ m n, 1 < m → m ≤ n → (f n : ennreal) ≤ f m :=
+      λ m n hm hmn, ennreal.coe_le_coe.2 (hf (zero_lt_one.trans hm) hmn),
+    simpa [h, ennreal.add_eq_top] using (ennreal.tsum_condensed_le hf) },
+  { replace hf : ∀ m n, 0 < m → m ≤ n → (f n : ennreal) ≤ f m :=
+      λ m n hm hmn, ennreal.coe_le_coe.2 (hf hm hmn),
+    simpa [h, ennreal.add_eq_top] using (ennreal.le_tsum_condensed hf) }
+end
+
+end nnreal
+
+/-- Cauchy condensation test for series of nonnegative real numbers. -/
+lemma summable_condensed_iff_of_nonneg {f : ℕ → ℝ} (h_nonneg : ∀ n, 0 ≤ f n)
+  (h_mono : ∀ ⦃m n⦄, 0 < m → m ≤ n → f n ≤ f m) :
+  summable (λ k : ℕ, (2 ^ k) * f (2 ^ k)) ↔ summable f :=
+begin
+  set g : ℕ → ℝ≥0 := λ n, ⟨f n, h_nonneg n⟩,
+  have : f = λ n, g n := rfl,
+  simp only [this],
+  have : ∀ ⦃m n⦄, 0 < m → m ≤ n → g n ≤ g m := λ m n hm h, nnreal.coe_le_coe.2 (h_mono hm h),
+  exact_mod_cast nnreal.summable_condensed_iff this
+end
+
+open real
+
+/-- Test for congergence of the `p`-series: the real-valued series `∑' n : ℕ, 1 / n ^ p` converges
+if and only if `1 < p`. -/
+@[simp] lemma summable_nat_rpow_inv {p : ℝ} : summable (λ n, (n ^ p)⁻¹ : ℕ → ℝ) ↔ 1 < p :=
+begin
```

Comment:
It would be nice to include a comment explaining the proof strategy, giving the formula for the condensed series and mentioning geometric series.